File: frontend/src/features/coordinacion/types/encuentros.ts
export interface SlotEncuentro {
  id: string
  materia_id: string
  titulo: string
  hora: string
  dia_semana: string | null
  fecha_inicio: string | null
  cant_semanas: number | null
  fecha_unica: string | null
  meet_url: string | null
  vig_desde: string
  vig_hasta: string | null
  instancias?: InstanciaEncuentro[]
}

export interface InstanciaEncuentro {
  id: string
  slot_id: string | null
  fecha: string
  hora: string
  titulo: string
  estado: string
  meet_url: string | null
  video_url: string | null
  comentario: string | null
}

export interface SlotPayload {
  materia_id: string
  titulo: string
  hora: string
  dia_semana?: string | null
  fecha_inicio?: string | null
  cant_semanas?: number | null
  fecha_unica?: string | null
  meet_url?: string | null
  vig_desde: string
  vig_hasta?: string | null
}

export interface InstanciaEditPayload {
  estado?: string
  meet_url?: string | null
  video_url?: string | null
  comentario?: string | null
}

File: backend/app/schemas/encuentros.py
"""Pydantic schemas for Encuentros (SlotEncuentro, InstanciaEncuentro)."""

from datetime import date

from pydantic import BaseModel, ConfigDict


class SlotEncuentroCreate(BaseModel):
    model_config = ConfigDict(extra="forbid")

    asignacion_id: str
    materia_id: str
    titulo: str
    hora: str
    dia_semana: str | None = None
    fecha_inicio: date | None = None
    cant_semanas: int | None = None
    fecha_unica: date | None = None
    meet_url: str | None = None
    vig_desde: date
    vig_hasta: date


class SlotEncuentroRead(BaseModel):
    model_config = ConfigDict(extra="forbid")

    id: str
    asignacion_id: str
    materia_id: str
    titulo: str
    hora: str
    dia_semana: str | None = None
    fecha_inicio: date | None = None
    cant_semanas: int | None = None
    fecha_unica: date | None = None
    meet_url: str | None = None
    vig_desde: date
    vig_hasta: date
    tenant_id: str


class InstanciaEncuentroUpdate(BaseModel):
    model_config = ConfigDict(extra="forbid")

    estado: str | None = None
    meet_url: str | None = None
    video_url: str | None = None
    comentario: str | None = None


class InstanciaEncuentroRead(BaseModel):
    model_config = ConfigDict(extra="forbid")

    id: str
    slot_id: str | None = None
    materia_id: str
    fecha: date
    hora: str
    titulo: str
    estado: str
    meet_url: str | None = None
    video_url: str | None = None
    comentario: str | None = None
    tenant_id: str
